### PR TITLE
fix(set-due-date): small phone layout 

### LINKED ISSUE #16200: In "Set due date" dialog, label and border line of input boxes can overlap
### Checked for duplicates?

- [X] This issue is not a duplicate

### Does it also happen in the desktop version?

- [X] This bug does not occur in the latest version of Anki

### What are the steps to reproduce this bug?

0. Use a recent repository
1. Open "Card browser"
2. Long press a card
3. Three-dots menu > "Set due date"
4. Tap the left tab

### Expected behaviour

The center `–` has sufficient padding:

https://github.com/ankidroid/Anki-Android/pull/15699

### Actual behaviour

In my device, the center `–` doesn't have sufficient padding:

### Debug info

```text
AnkiDroid Version = 2.18alpha8-debug (910795a5cb1caa26b121b3d85cab32d4cc2e1b76)

Backend Version = 0.1.36-anki24.04 (24.04 429bc9e14cefb597646a0e1beac6ef140f226b6f)

Android Version = 11 (SDK 30)

ProductFlavor = amazon

Manufacturer = OPPO

Model = CPH2013

Hardware = qcom

Webview User Agent = Mozilla/5.0 (Linux; Android 11; CPH2013 Build/RKQ1.200903.002; wv) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/123.0.6312.118 Mobile Safari/537.36

ACRA UUID = f74d8c3b-b173-4337-ae0c-819d3934bd5e

FSRS Enabled = false

Crash Reports Enabled = false
```

### (Optional) Anything else you want to share?

N/A

### Research

- [X] I am reporting a bug specific to AnkiDroid (Android app)
- [X] I have checked the [manual](https://ankidroid.org/docs/manual.html) and the [FAQ](https://github.com/ankidroid/Anki-Android/wiki/FAQ) and could not find a solution to my issue
- [X] (Optional) I have confirmed the issue is not resolved in the latest alpha release ([instructions](https://docs.ankidroid.org/manual.html#betaTesting))

### PR DESCRIPTION
**Pull request by david-allison**
## Purpose / Description

image

## Fixes
* Fixes #16200

## Approach
* Define `layout-w340dp` where the EditTexts are stacked vertically

## How Has This Been Tested?
Screenshot 2024-04-18 at 16 41 09

## Learning (optional, can help others)
320dp was insufficient, 340 seems fine

## Checklist
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

**Comment by david-allison** [comment]
Optimization: we should remove 'comment' from translated files to save space

File: AnkiDroid/src/main/res/layout-w340dp/set_due_date_range.xml [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+        app:layout_constraintStart_toEndOf="@+id/date_range_start_layout"
+        app:layout_constraintTop_toTopOf="@+id/date_range_start_layout"
+        />
+
+    <com.google.android.material.textfield.TextInputLayout
+        android:id="@+id/date_range_end_layout"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_marginTop="8dp"
+        android:layout_marginEnd="8dp"
```
Comment (lukstbit) [inline AnkiDroid/src/main/res/layout-w340dp/set_due_date_range.xml:73]: I can see the effect in the UI, but why was this needed?

**Review by lukstbit — APPROVED** [review]
LGTM, works on a 3 inch screen :)

File: AnkiDroid/src/main/res/layout-w340dp/set_due_date_range.xml (hunk 1 above)
Comment (david-allison) [reply AnkiDroid/src/main/res/layout-w340dp/set_due_date_range.xml]: It wasn't, fixed in additional commit, thanks!

**Review by BrayanDSO — APPROVED** [review]

**Comment by david-allison** [comment]
Before:

* https://github.com/ankidroid/Anki-Android/pull/16229

**Comment by david-allison** [comment]
* https://github.com/ankidroid/Anki-Android/pull/16231

**Pull request merged**